File: cli/Sources/TuistServer/Models/CacheTechnology.swift
public enum CacheTechnology: String, Sendable {
    case `default` = "default"
    case kura = "kura"

    var queryValue: Operations.getCacheEndpoints.Input.Query.technologyPayload? {
        switch self {
        case .default:
            nil
        case .kura:
            .kura
        }
    }
}

File: gradle/src/main/kotlin/dev/tuist/gradle/CacheTechnology.kt
package dev.tuist.gradle

enum class CacheTechnology(val rawValue: String) {
    DEFAULT("default"),
    KURA("kura");

    val queryValue: String?
        get() = when (this) {
            DEFAULT -> null
            KURA -> rawValue
        }

    companion object {
        fun fromEnvironment(envProvider: (String) -> String? = { System.getenv(it) }): CacheTechnology {
            val value = envProvider("TUIST_KURA")
            return if (value.isNullOrBlank()) DEFAULT else KURA
        }
    }
}
